smart_card_connector_app/src/pcsc-api-jstocxxtest.js

```diff
@@ -1206,6 +1206,28 @@ goog.exportSymbol('testPcscApi', {
       assert(called);
       assertEquals(result.getErrorCode(), API.SCARD_E_INVALID_HANDLE);
     },
+
+    // Test that no unexpected exceptions occur when the C++ module is crashing
+    // while there's a connected PC/SC client.
+    'testCrashWithActiveClient': async function() {
+      const BAD_CONTEXT = 123;
+      await testController.setUpCppHelper(
+          'LoggingTestHelper', /*helperArgument=*/ {});
+      await launchPcscServer(/*initialDevices=*/[]);
+      // Make a random call, to make sure all lazily created per-client state
+      // gets initialized.
+      await client.api.SCardIsValidContext(BAD_CONTEXT);
+
+      // Trigger the C++ module crash.
+      try {
+        await testController.sendMessageToCppHelper(
+            'LoggingTestHelper', 'crash-via-check');
+      } catch (e) {
+        // This is expected branch - discard the exception.
```

Comment:
This catches any exception - how does this then test that no unexpected exception occurred?

Reply:
Fixed, thanks for pointing this out.